sympy/series/limits.py

```diff
@@ -168,6 +168,16 @@ def doit(self, **hints):
         if e.is_Order:
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
+        #Check to see if e is a simple power series, like (-1/2)**x
+        if e.is_Pow and len(e.free_symbols) == 1:
+            k = e.free_symbols.pop()
+            if k == z:
+                try:
+                    if (abs(e.base) < 1 and z0 is S.Infinity) or (abs(e.base) > 1 and z0 is S.NegativeInfinity):
```

Comment:
That's what the `try` statement does--if `z` or any non-`Number` expression is in the base, then trying to assess the truth of `abs(e.base) < 1' throws a `TypeError`, and the function just continues on.